src/java.base/share/classes/java/nio/file/FileTreeWalker.java

```diff
@@ -198,15 +198,11 @@ IOException ioeException() {
      * the walk is following sym links is not. The {@code canUseCached}
      * argument determines whether this method can use cached attributes.
      */
-    @SuppressWarnings("removal")
     private BasicFileAttributes getAttributes(Path file, boolean canUseCached)
         throws IOException
     {
         // if attributes are cached then use them if possible
-        if (canUseCached &&
-            (file instanceof BasicFileAttributesHolder) &&
-            (System.getSecurityManager() == null))
-        {
+        if (canUseCached && (file instanceof BasicFileAttributesHolder)) {
```

Comment:
Maybe you don't want to do it in this PR but this can use pattern matching and avoid casting to BasicFileAttribtuesHolder.